change documentation for Buf and mdspan

mehmetyusufoglu opened the pull request:
Simple additions to cheetsheet
- to solve an ambiguity in `alpaka::Buf` part
- to show setting and getting items of `mdspan`

mehmetyusufoglu committed:
Improve documentation for buf and mdspan

fwyzard commented on docs/source/basic/cheatsheet.rst:
```diff
@@ -109,7 +109,11 @@ Allocate a buffer in host memory
      // Use alpaka vector as a static array for the extents
-     Vec<Dim, Idx> extent = value;
+     // If Dim is DimInt<1>;
+     Vec<Dim, Idx> extent = scalarIdxValue;
```
to me this would be more clear as
```suggestion
     Vec<Dim, Idx> extent = value;
```
or
```suggestion
     Vec<DimInt<1>, Idx> extent = value;
```

fwyzard commented on docs/source/basic/cheatsheet.rst:
```diff
@@ -109,7 +109,11 @@ Allocate a buffer in host memory
+     // If Dim is DimInt<1>;
+     Vec<Dim, Idx> extent = scalarIdxValue;
+     // If Dim is DimInt<2>;
+     Vec<Dim, Idx> extent = {scalarIdxValue1,scalarIdxValue2};
```
to me this would be more clear as
```suggestion
     Vec<Dim, Idx> extent = {value1, value2};
```
or
```suggestion
     Vec<DimInt<2>, Idx> extent = {value1, value2};
```

mehmetyusufoglu commented:
@fwyzard  your suggestions (first options) are added, thanks.

psychocoderHPC commented:
I will try to reduce the cheatsheet changes that our cheatsheet is fitting to two pdf pages again

psychocoderHPC committed:
shorten cheatsheet

- change naming to reflect alpaka index order
- remove comments where code can say everything

psychocoderHPC commented:
we should add more content to make the cheatsheet three pages or we should take care to keep two pdf sides. The printed version looks bad in case we have only a few lines on an empty page

mehmetyusufoglu commented:
> we should add more content to make the cheatsheet three pages or we should take care to keep two pdf sides. The printed version looks bad in case we have only a few lines on an empty page

I see, thanks. I think cheat-sheet doesn't have to be one sheet, it is used online by people as a shorthand and concise description.
